### PR TITLE
Port background-util.js to TS

### PR DESCRIPTION
**Pull request by Thaoden**
This is just a small port from `background-util.js` to TypeScript.

### CODE REVIEW

File: src/window_background/backgroundUtil.ts [hunk 1]
```diff
@@ -105,23 +105,28 @@ export function getDateFormat(dateStr) {
   }
 }
 
-export function normaliseFields(iterator) {
-  if (typeof iterator == "object") {
-    return _.transform(iterator, function(result, value, key) {
-      let nkey =
-        typeof key == "string" ? key.replace(/List$/, "").toLowerCase() : key;
```
Comment (lusbenjamin) [inline src/window_background/backgroundUtil.ts]: This logic did not make it into the TS conversion. Was that a deliberate change?

Comment (Manuel-777) [reply src/window_background/backgroundUtil.ts]: This function was only useful in the payload decoder, Which was only used to get the sideboard changes on BO3, does that still work with these changes?

Comment (Thaoden) [reply src/window_background/backgroundUtil.ts]: TS types told me that `key` is always of type `number`, so the first part of the ternary didn't ever evaluate, thus I removed it.

Comment (Manuel-777) [reply src/window_background/backgroundUtil.ts]: pushed c5e5d933b714522a450eb3b15c4d7917ff412f49 with a "fix" for this, it makes TS happy without changing the functionality.

**Pull request merged**